spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -2763,6 +2765,15 @@ def keyPressEvent(self, event):
                     TextEditBaseWidget.keyPressEvent(self, event)
                     self.fix_indent(comment_or_string=cmt_or_str)
                     self.textCursor().endEditBlock()
+
+                    self.left_line(position, source_button=key)
+
+        elif key in [Qt.Key_Left, Qt.Key_Up, Qt.Key_Right, Qt.Key_Down]:

```

Comment:
This should include page up and page down as well.

Reply:
Does it also work for ``Ctrl-Left/Right/Up/Down/Home/End/PageUp/Pagedown`` as well?

Reply:
It doesn't it also won't work with moving the cursor with the outliner, runcell, taceback links or the upcoming bookmarks. This should really be tied into some cursor movement event instead of key based.

Reply:
@bcolsen Good point. We already track when the textcursor changes line in order to move the "Current line" syntax highlighting background color around, and that works correctly in all those scenarios. Therefore, I imagine you could just use whatever event that uses?